tests/utils/stl/test/format.py

```diff
@@ -232,9 +232,11 @@ def execute(self, test, litConfig):
 
                     report += stl.util.makeReport(cmd, out, err, rc)
                     if (step.shouldFail and rc == 0) or (not step.shouldFail and rc != 0):
-                        return (failVar, report)
+                        if someFail:
+                            test.xfails = ['*']

```

Comment:
does someFail mean that some of the tests in a batch will fail or that a specific test will sometimes fail?

if the first then does this oversuppress failing tests?

Reply:
`someFail` is meant to be true if the build step or the test step should fail but we're not sure which one. Therefore once it actually fails we suppress the failure.